### ISSUE TITLE
[Bug]: Terminal logger can leave terminal in progress state

### ISSUE THREAD

**Issue by ladipro** [dotnet/msbuild#9898]
### Issue Description

Minor annoyance but since we're enabling TL by default, I think it's worth a quick investigation.

### Steps to Reproduce

```
C:\src\msbuild> SET MSBUILDLIVELOGGER=auto
C:\src\msbuild> build.cmd
[...building...]
```

Hit Ctrl+C

```
Terminate batch job (Y/N)?
```

Reply Y or N, it does not make a difference.

### Expected Behavior

Terminal window exits progress state.

### Actual Behavior

Terminal window stays in progress state.

![image](https://github.com/dotnet/msbuild/assets/12206368/71a59336-c850-4e55-aa74-6f77cb9c59ff)

### Versions & Configurations

MSBuild version 17.10.0-preview-24164-02+a7c9b7587 for .NET Framework

**Comment by rainersigwald** [comment]
Yeah this has been bugging me too, thanks for filing.

We should see if we can emit the return-to-normal message on cancellation. We should know if the TL would have been used at an appropriate level (though in general I don't think all loggers get enough info to do this).

**Comment by MichalPavlik** [comment]
It should be fixed by #10055.
